src/main/java/net/minecraftforge/client/model/generators/BlockModelBuilder.java

```diff
@@ -16,9 +25,291 @@
+            return this;
+        }
+
+        /**
+         * Sets the right rotation of the root transform.
+         *
+         * @param postRotation the right rotation
+         * @return this builder
+         * @throws NullPointerException if {@code rightRotation} is {@code null}
+         */
+        public RootTransformBuilder postRotation(Quaternion postRotation)
+        {
+            return rightRotation(postRotation);
+        }
+
+        /**
+         * Sets the scale of the root transform.
+         *
+         * @param scale the scale
+         * @return this builder
+         * @throws NullPointerException if {@code scale} is {@code null}
+         */
+        public RootTransformBuilder scale(float scale)
+        {
+            return scale(new Vector3f(scale, scale, scale));
+        }
+
+        /**
+         * Sets the scale of the root transform.
+         *
+         * @param scale the scale vector
+         * @return this builder
+         * @throws NullPointerException if {@code scale} is {@code null}
+         */
+        public RootTransformBuilder scale(Vector3f scale)
+        {
+            this.scale = Preconditions.checkNotNull(scale, "Scale must not be null");
+            return this;
+        }
+
+        /**
+         * Sets the root transform.
+         *
+         * @param transformation the transformation to use
+         * @return this builder
+         * @throws NullPointerException if {@code transformation} is {@code null}
+         */
+        public RootTransformBuilder transform(Transformation transformation)
+        {
+            Preconditions.checkNotNull(transformation, "Transformation must not be null");
+            this.translation = transformation.getTranslation();
+            this.leftRotation = transformation.getLeftRotation();
+            this.rightRotation = transformation.getRightRotation();
+            this.scale = transformation.getScale();
+            return this;
+        }
+
+        /**
+         * Sets the origi of the root transform.
+         *
+         * @param origin the origin vector
+         * @return this builder
+         * @throws NullPointerException if {@code origin} is {@code null}
+         */
+        public RootTransformBuilder origin(Vector3f origin)
+        {
+            this.originVec = Preconditions.checkNotNull(origin, "Origin must not be null");
+            this.origin = null;
+            return this;
+        }
+
+        /**
+         * Sets the origin of the root transform.
+         *
+         * @param origin the origin name
+         * @return this builder
+         * @throws NullPointerException if {@code origin} is {@code null}
+         * @throws IllegalArgumentException if {@code origin} is not {@code center}, {@code corner} or {@code opposing-corner}
+         */
+        public RootTransformBuilder origin(TransformOrigin origin)
+        {
+            this.origin = Preconditions.checkNotNull(origin, "Origin must not be null");;
```

Comment:
Remove the unnecessary semicolon.
```suggestion
            this.origin = Preconditions.checkNotNull(origin, "Origin must not be null");
```